modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -349,10 +355,14 @@ public void onException(ConnectionException exception) {
       return;
     }
 
-    muleContext.getExceptionListener().handleException(exception, getLocation());
-
-    refreshTokenIfNecessary(getConfigurationInstance()
-        .flatMap(configurationInstance -> configurationInstance.getConnectionProvider()).orElse(null), exception);
+    boolean accessTokenExpired = refreshTokenIfNecessary(
+                                                         getConfigurationInstance()
+                                                             .flatMap(ConfigurationInstance::getConnectionProvider).orElse(null),
+                                                         exception);
+    boolean shouldHandleException = !accessTokenExpired || SHOULD_HANDLE_ACCESS_TOKEN_EXPIRED_EXCEPTIONS_ON_SOURCES;
+    if (shouldHandleException) {
+      muleContext.getExceptionListener().handleException(exception, getLocation());
```

Comment:
this is the wrong place to place this if. This is not just preventing the notification from being fired, this is skipping the exception handler alltogether, which is doing way more things than simply firing the notification